content/blog/2018-09-04-birds-taxo-traits.md

```diff
@@ -0,0 +1,559 @@
+  }
+  
+  common_names <- purrr::map_chr(order_members, translate)
+  
+  species <-  stringr::str_wrap(toString(common_names),width = 50)
+  
+  tibble::tibble(id = id,
+                 order = order,
+                 size = length(order_members),
+                 species = species)
+  
+}
+
+orders <- purrr::map_df(unique(tree$classification$order),
+                        find_order_node_id,
+                        p4)
+
+str(orders)
+```
+
+    ## Classes 'tbl_df', 'tbl' and 'data.frame':    18 obs. of  4 variables:
+    ##  $ id     : num  237 238 235 239 231 10 228 226 224 222 ...
+    ##  $ order  : Factor w/ 18 levels "Anseriformes",..: 3 12 13 1 5 4 15 11 16 17 ...
+    ##  $ size   : int  32 86 9 35 5 1 7 6 5 3 ...
+    ##  $ species: chr  "Black-headed Gull, Common Tern, Northern Lapwing,\nYellow-legged Gull, Common Sandpiper, Eurasian\nCurlew, Mew "| __truncated__ "Carrion Crow, Eurasian Magpie, House Sparrow,\nShort-toed Treecreeper, Eurasian Blackbird,\nEuropean Greenfinch"| __truncated__ "Gray Heron, Great Cormorant, Great Egret, Eurasian\nSpoonbill, Purple Heron, Cattle Egret, Little\nBittern, Lit"| __truncated__ "Mallard, Mute Swan, Common Goldeneye, Common\nMerganser, Common Pochard, Ferruginous Duck,\nGreen-winged Teal, "| __truncated__ ...
+
+For each order, I’ll get a silhouette from Phylopic using [Scott
+Chamberlain’s `rphylopic` package](https://github.com/sckott/rphylopic).
+
+``` r
+get_results <- function(name){
+  
+  id <- rphylopic::name_search(name)
+  rphylopic::name_images(id$canonicalName[1,1])
+}
+
+get_pic <- function(order, classification){
+  message(order)
+  
+  # shortcurt for flamingos
+  if(order == "Phoenicopteriformes"){
+    return(tibble::tibble(pic_id = "28473411-c079-4654-bbb7-34a5615bb414",
+                          order = "Phoenicopteriformes"))
+  }
+  
+  classification <- classification[classification$order == order,]
+  
+  results <- get_results(order)
+  
+  if (length(results$same) > 0){
+    # best case
+    pic_id <- results$`same`[[1]]$`uid`
+  }else{
+    # take the most common species
+    # and get any pic of it
+    results <- get_results(classification$species[
+      classification$n == max(classification$n, na.rm=TRUE)
+    ][1])
+    results <- purrr::keep(results, function(x) length(x) > 0)
+    results <- unlist(results)
+    results <- results[length(results)]
+    pic_id <- as.character(results)
+  }
+  
+  tibble::tibble(pic_id = pic_id,
+                 order = order)
+}
+
+
+library("magrittr")
+classification <- tree$classification %>%
+  dplyr::mutate(species = as.character(species)) %>%
+  dplyr::left_join(abundance,
+                   by = c("species" = "scientific_name"))
+
+ids <- purrr::map_df(orders$order, get_pic,          classification)
+save(ids, file = file.path("taxo", "ids.RData"))
+```
+
+It is rather tricky to automatically get pics from Phylopic since you
+might not get one for the order itself, maybe one for the subtaxa
```

Comment:
should it be "subtaxon" in "maybe one for the subtaxa instead"?

Reply:
indeed!